src/main/java/com/siit/hospital_manager/controller/DoctorController.java

```diff
@@ -0,0 +1,130 @@
+package com.siit.hospital_manager.controller;
+
+
+import com.siit.hospital_manager.model.dto.*;
+import com.siit.hospital_manager.service.DoctorService;
+import com.siit.hospital_manager.service.SpecialisationService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+
+import java.util.List;
+
+import static com.siit.hospital_manager.util.AuthUtils.isAdmin;
+import static com.siit.hospital_manager.util.AuthUtils.isPatient;
+
+
+@Controller
+@RequiredArgsConstructor
+@RequestMapping("/mvc/doctor")
+public class DoctorController {
+
+    private final DoctorService doctorService;
+    private final SpecialisationService specialisationService;
+
+    @GetMapping("/viewAll")
+    public String viewAll(Model model, Authentication authentication){
+        List<DoctorDto> doctorsList = doctorService.findAll();
+        model.addAttribute("doctors", doctorsList);
+        if (isAdmin(authentication)) {
+            List<SpecialisationDto> specialisationList = specialisationService.findAll();
+            model.addAttribute("specialisations", specialisationList);
+            return "admin/viewAllDoctors";
+        }
+        else {
+            return "doctor/viewAll";
+        }
+    }
+
+    @GetMapping("/viewDoctorsBySpecialisation/{specialisationId}")
+    public String viewDoctorsBySpecialisation(Model model, @PathVariable("specialisationId") Integer specialisationId, Authentication authentication) {
+        List<DoctorDto> doctorsList = doctorService.findAllBySpecialisation(specialisationService.findSpecialisationById(specialisationId));
```

Comment:
this could be in a service and also treat cases in which specialisation is not found